dace/runtime/include/dace/perf/reporting.h

```diff
@@ -68,11 +68,23 @@ namespace perf {
             const char *cat,
             const char *counter_name,
             unsigned long int counter_val
+        ) {
+            add_counter(name, cat, counter_name, counter_val, -1, -1, -1, -1);
```

Comment:
no, this just removes tid. We need it for normal counters (e.g., for tasklets inside a map).
Please return tid computation here (if add_counter is called without a tid).

Reply:
Hm, I see, but how to support this thread_id datatype and the integers from omp_get_thread_num()?

Reply:
Upon further analysis of this section, I agree. You can replace the `-1` thread ID with `std::this_thread::get_id()` in this overload call where no other thread ID is provided, no?

Reply:
std::this_thread::get_id() != int ? What to store in the EventTrace struct?

Reply:
You can `std::hash<std::thread::id>(mytid)` it into an integer, afaik.

Reply:
See https://stackoverflow.com/questions/7432100/how-to-get-integer-thread-id-in-c11

Reply:
Haha yeah, saw the same stackoverflow discussion, but wasn't sure whether this is actually safe/good to do